projects/core/src/auth/user-auth/http-interceptors/auth.interceptor.spec.ts

```diff
@@ -0,0 +1,272 @@
+import {
+  HttpClient,
+  HttpParams,
+  HttpRequest,
+  HttpUserEvent,
+  HTTP_INTERCEPTORS,
+} from '@angular/common/http';
+import {
+  HttpClientTestingModule,
+  HttpTestingController,
+  TestRequest,
+} from '@angular/common/http/testing';
+import { TestBed } from '@angular/core/testing';
+import { Observable, of, Subscription } from 'rxjs';
+import { AuthToken } from '../models/auth-token.model';
+import { AuthConfigService } from '../services/auth-config.service';
+import { AuthHeaderService } from '../services/auth-header.service';
+import { AuthInterceptor } from './auth.interceptor';
+
+class MockAuthHeaderService implements Partial<AuthHeaderService> {
+  alterRequest(req) {
+    return req;
+  }
+  shouldCatchError() {
+    return true;
+  }
+  handleExpiredAccessToken() {
+    return of() as Observable<HttpUserEvent<AuthToken>>;
+  }
+  handleExpiredRefreshToken() {}
+}
+
+class MockAuthConfigService implements Partial<AuthConfigService> {
+  getTokenEndpoint() {
+    return '/authorizationserver/token';
+  }
+}
+
+describe('AuthInterceptor', () => {
+  let httpMock: HttpTestingController;
+  let http: HttpClient;
+  let authHeaderService: AuthHeaderService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [
+        { provide: AuthHeaderService, useClass: MockAuthHeaderService },
+        { provide: AuthConfigService, useClass: MockAuthConfigService },
+        {
+          provide: HTTP_INTERCEPTORS,
+          useClass: AuthInterceptor,
+          multi: true,
+        },
+      ],
+    });
+
+    httpMock = TestBed.inject(HttpTestingController);
+    authHeaderService = TestBed.inject(AuthHeaderService);
+    http = TestBed.inject(HttpClient);
+  });
+
+  it(`Should operate on request returned from AuthHeaderService alterRequest method`, () => {
+    spyOn(authHeaderService, 'alterRequest').and.returnValue(
+      new HttpRequest('GET', '/test')
+    );
+
+    const sub: Subscription = http.get('/xxx').subscribe((result) => {
+      expect(result).toBeTruthy();
+    });
+
+    const mockReq: TestRequest = httpMock.expectOne((req) => {
+      return req.method === 'GET' && req.url === '/test';
+    });
+
+    mockReq.flush('someData');
+    sub.unsubscribe();
+  });
+
+  it(`Should handle 401 error for expired token occ calls`, () => {
+    spyOn(
+      authHeaderService,
+      'handleExpiredAccessToken'
+    ).and.callFake((_, next) => next.handle(new HttpRequest('GET', '/test')));
+    const sub: Subscription = http.get('/occ').subscribe((result) => {
+      expect(result).toEqual('someText');
```

Comment:
I'm suspcious with expects in subsribe without additoinal check like `done()`, because there is high risk that in case of no emission, they won't execute at all.
It applies to more cases in this file and in tokenRevocationInterceptor.